indexes/plugin.js

```diff
@@ -143,11 +143,11 @@ module.exports = class Plugin {
     const subClassReset = this.reset
     this.reset = (cb) => {
       if (subClassReset) subClassReset.call(this)
+      this.batch = []
```

Comment:
this batch reset is def. correct in that those are so to speak in flight, so we need to clear them before clearing level.